src/containers/survey/HourlyAppUsageSurvey.js

```diff
@@ -173,21 +173,21 @@ const HourlyAppUsageSurvey = (props :Props) => {
     isInstructionsModalVisible
   } = state;
 
-  const createSubmissionData = () => {
+  useEffect(() => {
+    const createSubmissionData = () => {
```

Comment:
No change required. An alternative would be to move this logic into the saga, where you can have more robust data processing all surrounded by a try-catch in case something is wrong. Passing a callback as a param is not ideal and we tend to avoid that pattern.